src/components/BookItem.tsx

```diff
@@ -56,25 +60,32 @@ export const BookItem = ({
       className={`result-card ${isDetailSelected ? 'selected' : ''} ${isSelected ? 'checked' : ''}`}
       onClick={handleBookClick}
     >
-      <label className="book-checkbox-container" htmlFor={`book-${book.key}`}>
+      <label
+        className="book-checkbox-container"
+        htmlFor={`book-${book.key}`}
+        onClick={(e) => e.stopPropagation()}
+      >
         <input
           id={`book-${book.key}`}
           type="checkbox"
           checked={isSelected}
-          onChange={handleCheckboxChange}
+          onClick={(e) => e.stopPropagation()}
+          onChange={(e) => {
```

Comment:
```suggestion
         onChange={handleCheckboxChange}
```
e.stopPropagation() уже вызывается внутри handleCheckboxChange